File: tanklaboratory/src/main/java/com/skx/tomike/tanklaboratory/widget/activity/ViewFocusActivity.java
package com.skx.tomike.tanklaboratory.widget.activity;

import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;

import com.skx.tomike.tanklaboratory.R;
import com.skx.tomikecommonlibrary.base.BaseViewModel;
import com.skx.tomikecommonlibrary.base.SkxBaseActivity;
import com.skx.tomikecommonlibrary.base.TitleConfig;

/**
 * 描述 : view 焦点
 * 作者 : shiguotao
 * 版本 : V1
 * 创建时间 : 2020/4/1 7:46 PM
 */
public class ViewFocusActivity extends SkxBaseActivity<BaseViewModel> {

    @Override
    protected void initParams() {
    }

    @Override
    protected TitleConfig configHeaderTitle() {
        return new TitleConfig.Builder().setTitleText("EditText 焦点").create();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_view_focus;
    }

    @Override
    protected void initView() {
        EditText mEtName = findViewById(R.id.et_editTextFocus_name);
        mEtName.setOnFocusChangeListener(focusChangeListener);
        mEtName.setTag("姓名输入框EditText");

        EditText mEtNickname = findViewById(R.id.et_editTextFocus_nickname);
        mEtNickname.setOnFocusChangeListener(focusChangeListener);
        mEtNickname.setTag("昵称输入框EditText");

        RadioButton mRbtnMan = findViewById(R.id.rbtn_editTextFocus_sex_man);
        mRbtnMan.setOnFocusChangeListener(focusChangeListener);
        mRbtnMan.setTag("性别单选框男RadioButton");

        RadioButton mRbtnWomen = findViewById(R.id.rbtn_editTextFocus_sex_women);
        mRbtnWomen.setOnFocusChangeListener(focusChangeListener);
        mRbtnWomen.setTag("性别单选框女RadioButton");

        TextView mTvText = findViewById(R.id.btn_editTextFocus_text);
        mTvText.setOnFocusChangeListener(focusChangeListener);
        mTvText.setTag("纯文本TextView");
        mTvText.setFocusable(true);
        mTvText.setFocusableInTouchMode(true);
        mTvText.requestFocus();

        Button mBtnSubmit = findViewById(R.id.btn_editTextFocus_submit);
        mBtnSubmit.setOnFocusChangeListener(focusChangeListener);
        mBtnSubmit.setTag("提交按钮button");

        mBtnSubmit.setFocusable(true);
        mBtnSubmit.setFocusableInTouchMode(true);
    }

    private View.OnFocusChangeListener focusChangeListener = new View.OnFocusChangeListener() {
        @Override
        public void onFocusChange(View v, boolean hasFocus) {
            Log.e(TAG, v.getTag() + " -> " + (hasFocus ? "获得焦点" : "失去焦点"));
        }
    };
}

File: tank/src/main/java/com/skx/tomike/tank/widget/activity/ViewFocusActivity.kt
package com.skx.tomike.tank.widget.activity

import android.util.Log
import android.view.View
import android.view.View.OnFocusChangeListener
import android.widget.*
import com.alibaba.android.arouter.facade.annotation.Route
import com.skx.common.base.BaseViewModel
import com.skx.common.base.SkxBaseActivity
import com.skx.common.base.TitleConfig
import com.skx.tomike.tank.R
import com.skx.tomike.tank.ROUTE_PATH_VIEW_FOCUS

/**
 * 描述 : view 焦点
 * 作者 : shiguotao
 * 版本 : V1
 * 创建时间 : 2020/4/1 7:46 PM
 */
@Route(path = ROUTE_PATH_VIEW_FOCUS)
class ViewFocusActivity : SkxBaseActivity<BaseViewModel<*>>() {

    override fun initParams() {}

    override fun configHeaderTitle(): TitleConfig {
        return TitleConfig.Builder().setTitleText("EditText 焦点").create()
    }

    override fun layoutId(): Int {
        return R.layout.activity_view_focus
    }

    override fun initView() {
        findViewById<EditText>(R.id.et_viewFocus_name).apply {
            tag = "姓名输入框EditText"
            onFocusChangeListener = mFocusChangeListener
        }
        findViewById<EditText>(R.id.et_viewFocus_nickname).apply {
            tag = "昵称输入框EditText"
            onFocusChangeListener = mFocusChangeListener
        }
        findViewById<RadioGroup>(R.id.rg_viewFocus_sex).apply {
            tag = "性别单选框Group"
            onFocusChangeListener = mFocusChangeListener
            setOnCheckedChangeListener { group, checkedId ->
                val child = group.findViewById<View>(checkedId)
                Log.e(TAG, (child.tag?.toString() ?: "") + " -> 触发了点击事件")
            }
        }
        findViewById<RadioButton>(R.id.rbtn_viewFocus_sexGroup_man).apply {
            tag = "性别单选框男RadioButton"
            onFocusChangeListener = mFocusChangeListener
        }
        findViewById<RadioButton>(R.id.rbtn_viewFocus_sexGroup_women).apply {
            tag = "性别单选框女RadioButton"
            onFocusChangeListener = mFocusChangeListener
        }
        findViewById<TextView>(R.id.tv_viewFocus_text).apply {
            tag = "纯文本TextView"
            onFocusChangeListener = mFocusChangeListener
            isFocusable = true
            isFocusableInTouchMode = true
            requestFocus()
            setOnClickListener {
                Log.e(TAG, (it.tag?.toString() ?: "") + " -> 触发了点击事件")
            }
        }

        findViewById<Button>(R.id.btn_viewFocus_submit).apply {
            tag = "提交按钮button"
            onFocusChangeListener = mFocusChangeListener
            setOnClickListener {
                Log.e(TAG, (it.tag?.toString() ?: "") + " -> 触发了点击事件")
            }
        }
    }

    private val mFocusChangeListener = OnFocusChangeListener { v, hasFocus ->
        Log.e(TAG, (v.tag?.toString() ?: "") + " -> " + if (hasFocus) "获得焦点" else "失去焦点")
    }
}